pages/studentInstanceQuestionExam/studentInstanceQuestionExam.js

```diff
@@ -114,14 +114,15 @@ router.post('/', function(req, res, next) {
 
 router.get('/', function(req, res, next) {
     if (res.locals.assessment.type !== 'Exam') return next();
-    const variant_id = null;
-    question.getAndRenderVariant(variant_id, res.locals, function(err) {
+    logPageView(req, res, (err) => {
         if (ERR(err, next)) return;
-        logPageView(req, res, (err) => {
+        question.getAndRenderVariant(variant_id, res.locals, function(err) {
             if (ERR(err, next)) return;
             res.render(__filename.replace(/\.js$/, '.ejs'), res.locals);
         });
     });
+    const variant_id = null;
+
```

Comment:
This seems wrong now? We need to pass `variant_id = null` to `getAndRenderVariant()` to tell it to generate a new variant.

Also, we need to do `logPageView()` after `getAndRenderVariant()` because `getAndRenderVariant()` is going to add the new variant to `res.locals` from where `logPageView()` will read the `variant_id` for the page view log. Pretty nasty, I know!

Reply:
Mutability has its perks, but man, these global context objects are a mess :)